tap_square/client.py

```diff
@@ -295,3 +302,46 @@ def get_payments(self, start_time, bookmarked_cursor):
                 raise Exception(result.errors)
 
             yield (result.body.get('payments', []), result.body.get('cursor'))
+
+
+    def get_batch_token(self, link): #pylint: disable=no-self-use
+        if link:
+            url = link[link.find('<')+1:link.find('>')]
+            parsed = urlparse.urlparse(url)
+            batch_token = parse_qs(parsed.query)['batch_token'][0]
+            return int(batch_token)
+        return None
```

Comment:
Can we change the order here to do if not link return None first? Like validate the input. Then do the url and batch token generation in the non-branch code?

Reply:
Can we not use `find`? `requests` has utils functions for stuff like this